elide-integration-tests/src/test/java/com/yahoo/elide/tests/ResourceIT.java

```diff
@@ -443,72 +443,61 @@ public void testReadPermissionWithFilterCheckCollectionId() {
-                .extract().response().asString();
+                .body(equalTo(expected3));
 
         //test authentication fail querying with ID == 3
-        String getResult4 = given()
+        given()
                 .contentType(JSONAPI_CONTENT_TYPE)
                 .accept(JSONAPI_CONTENT_TYPE)
                 .get("/filterExpressionCheckObj/3")
                 .then()
-                .statusCode(HttpStatus.SC_NOT_FOUND)
-                .extract().response().asString();
+                .statusCode(HttpStatus.SC_NOT_FOUND);
 
         //test authentication pass query a relation of object
-        String getResult5 = given()
+        String expected5 = "{\"data\":[{\"type\":\"anotherFilterExpressionCheckObj\",\"id\":\"1\",\"attributes\":{\"anotherName\":\"anotherObj1\",\"createDate\":1999},\"relationships\":{\"linkToParent\":{\"data\":[{\"type\":\"filterExpressionCheckObj\",\"id\":\"1\"}]}}}]}";
+        given()
                 .contentType(JSONAPI_CONTENT_TYPE)
                 .accept(JSONAPI_CONTENT_TYPE)
                 .get("/filterExpressionCheckObj/1/listOfAnotherObjs")
                 .then()
                 .statusCode(HttpStatus.SC_OK)
-                .extract().response().asString();
+                .body(equalTo(expected5));
 
         //test authentication pass query a relation of object
-        String getResult6 = given()
+        given()
                 .contentType(JSONAPI_CONTENT_TYPE)
                 .accept(JSONAPI_CONTENT_TYPE)
                 .get("/anotherFilterExpressionCheckObj")
                 .then()
                 .statusCode(HttpStatus.SC_OK)
-                .extract().response().asString();
-
-        String expected1 = "{\"data\":[{\"type\":\"filterExpressionCheckObj\",\"id\":\"1\",\"attributes\":{\"name\":\"obj1\"},\"relationships\":{\"listOfAnotherObjs\":{\"data\":[{\"type\":\"anotherFilterExpressionCheckObj\",\"id\":\"1\"}]}}},{\"type\":\"filterExpressionCheckObj\",\"id\":\"2\",\"relationships\":{\"listOfAnotherObjs\":{\"data\":[]}}}]}";
-
-        String expected2 = "{\"data\":{\"type\":\"filterExpressionCheckObj\",\"id\":\"1\",\"attributes\":{\"name\":\"obj1\"},\"relationships\":{\"listOfAnotherObjs\":{\"data\":[{\"type\":\"anotherFilterExpressionCheckObj\",\"id\":\"1\"}]}}}}";
-
-        String expected3 = "{\"data\":{\"type\":\"filterExpressionCheckObj\",\"id\":\"2\",\"relationships\":{\"listOfAnotherObjs\":{\"data\":[]}}}}";
-
-        String expected5 = "{\"data\":[{\"type\":\"anotherFilterExpressionCheckObj\",\"id\":\"1\",\"attributes\":{\"anotherName\":\"anotherObj1\",\"createDate\":1999},\"relationships\":{\"linkToParent\":{\"data\":[{\"type\":\"filterExpressionCheckObj\",\"id\":\"1\"}]}}}]}";
-
-        assertEquals(getResult1, expected1);
-        assertEquals(getResult2, expected2);
-        assertEquals(getResult3, expected3);
-        assertEquals(getResult5, expected5);
-        assertEquals(getResult6, expected5);
+                .body(equalTo(expected5));
```

Comment:
expected6?

Reply:
No, that is how it worked.
```java
        assertEquals(getResult5, expected5);
        assertEquals(getResult6, expected5);
```

Reply:
I assume it is making 2 different requests expecting the same results.